orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/gce/GoogleImageTagger.java

```
@@ -0,0 +1,192 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.gce;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.clouddriver.tasks.image.ImageTagger;
+import com.netflix.spinnaker.orca.clouddriver.utils.CloudProviderAware;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+@Component
+public class GoogleImageTagger implements ImageTagger, CloudProviderAware {
+  private static final Logger log = LoggerFactory.getLogger(GoogleImageTagger.class);
+  private static final Set<String> BUILT_IN_TAGS = new HashSet<>(Arrays.asList("appversion", "build_host"));
+
+  @Autowired
+  OortService oortService;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @Override
+  public ImageTagger.OperationContext getOperationContext(Stage stage) {
+    StageData stageData = (StageData) stage.mapTo(StageData.class);
+
+    Collection<MatchedImage> matchedImages = findImages(stageData.imageNames, stage);
+
+    stageData.imageNames = matchedImages.stream()
+      .map(matchedImage -> matchedImage.imageName)
+      .collect(Collectors.toSet());
+
+    // Built-in tags are not updatable
+    Map<String, String> tags = stageData.tags.entrySet().stream()
+      .filter(entry -> !BUILT_IN_TAGS.contains(entry.getKey()))
```

Mainly going for consistency with the aws implementation of image tagger here. We could handle it in either spot, but if done in the validator it would throw an error instead of just ignoring it. That list of built-in tags is 'known' by orca/clouddriver/rosco/deck.